### PR TITLE
Run tests on Node.js 20

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Node 20 has been releases so adds it to test matrix.

ref: https://openjsf.org/blog/2023/04/18/node-js-20-now-available/

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Maybe we should add a comment to explain how we test.

- linux: every version
- mac/windows: lowest/highest-even version
- Full test: highest-even version on linux

**Comment by kachkaev** [comment]
How about running full tests on the latest available LTS version instead? This would include coverage. Node 20 may be a bit bumpy for the next few months till it becomes LTS.

**Comment by sosukesuzuki** [comment]
I'm 👍 for @kachkaev opinion.

so

linux: every version
mac/windows: lowest/highest-even version
Full test: LTS version on linux

**Comment by fisker** [comment]
I disagree. 

`FULL_TEST` only run extra comparison, it's pure JS, no extra Node.js features required, we can run it on any version already included, we only need it should be fast (the latest version normally faster, not always though).
https://github.com/prettier/prettier/blob/b232005022b063a7a3b8da2c87b94281a70188bb/.github/workflows/dev-test.yml#L33


I don't know how the matrix you expect to use. But if we use LTS version, there will be extra maintenance cost.

If we use

```
- "lts/*" (with FULL_TEST=1)
- "20"
- "18"
- "16"
````

There will be extra job to run. since "lts/*" and `"16"` is the same.


If we use

```
- "20"
- "18"
- "16" (with FULL_TEST=1)
````

We'll need move `FULL_TEST=1` when `16` is not active.

**Comment by fisker** [comment]
Okay, I changed the matrix, removed `exclude`, should be easier to setup.

I can accept

```diff
        include:
          - os: "ubuntu-latest"
-            # Latest even version
+            # LTS version
-            node: "20"
+            node: "18"
            FULL_TEST: true
-          # Versions between minimal version and latest even version
+          # Versions not tested on ubuntu
          # If latest version is an odd number, should be listed bellow too
          - os: "ubuntu-latest"
-            node: "18"
+            node: "20"
          - os: "ubuntu-latest"
            node: "16"
```

If you insistent.

**Comment by sosukesuzuki** [comment]
I don't mean to be so insistent. I don't have a strong preference, so Fisker's policy is fine to me.

**Comment by kachkaev** [comment]
Note that there can be two supported LTS versions at a given time:
https://github.com/nodejs/Release

I guess we want to run full tests on a supported LTS version with the highest version number. E.g. in Nov 2023 it would be Node 20 eventhough node 18 will be still LTS at that time.

**Comment by fisker** [comment]
So, you want this change https://github.com/prettier/prettier/pull/14721#issuecomment-1514053682

**Comment by fisker** [comment]
> I guess we want to run full tests on a supported LTS version with the highest version number. 

No, I don't want to update it (we can change it when we add/drop versions), because it doesn't matter what version is running full tests, as long as it's not droped. https://github.com/prettier/prettier/pull/14721#issuecomment-1514033254

**Review by sosukesuzuki — COMMENTED** [review]
LGTM

File: .github/workflows/dev-test.yml [hunk 1]
```diff
@@ -26,26 +26,26 @@ jobs:
...
+          # Versions not tested on linux, normally only even versions
+          # If latest version is an odd number, it can be listed bellow too
+          # - os: "ubuntu-latest"
+          #   node: "20"
     env:
       ENABLE_CODE_COVERAGE: ${{ matrix.ENABLE_CODE_COVERAGE }}
       FULL_TEST: ${{ matrix.FULL_TEST }}
       CHECK_TEST_PARSERS: ${{ matrix.CHECK_TEST_PARSERS }}
-    name: Node.js ${{ matrix.node }} on ${{ matrix.os }}
+    name: Node.js ${{ matrix.node }} on ${{ matrix.os }}${{ FULL_TEST && ' (Full Test)' || ''  }}
```
Comment (fisker) [inline .github/workflows/dev-test.yml:48]:
```suggestion
    name: Node.js ${{ matrix.node }} on ${{ matrix.os }}${{ matrix.FULL_TEST && ' (Full Test)' || '' }}
```

**Comment by sosukesuzuki** [comment]
Looks good

**Pull request merged**